### DISCUSSION TITLE
How to limit the maximum temperature

### DISCUSSION THREAD

**Discussion by czc-zju** [AMReX-Combustion/PeleC#617] [Q&A]
In my case, when the temperature exceeds 4000K, the calculation diverges. I would like to add an artificial limit. If the temperature is greater than 4000K, it is assigned to 3900K. Which file should I modify?

**Comment by drummerdoc** [comment]
I think you'll find that you'll end up chasing such a hack all over. One problem is that the eos is used to get T from (e,Y) and to get e from (T,Y).  Thus, you'll end up violating conservation of energy unless you put such a hack all over the place.  I've seen the astro folks that use the Castro code do similar things, and it permeates the code everywhere. Better in my mind to find why the temperature is climbing that high in the first place.  Are you under-resolving some chemistry in time, or is this perhaps happening on a covered coarse cell that is set via averaging very different states, etc.?  Can you narrow down the circumstances where this becomes an issue?
